src/vivarium_public_health/risks/data_transformations.py

```diff
@@ -417,6 +417,45 @@ def validate_distribution_data_source(builder, risk: EntityString):
     else:
         raise ValueError(f"Unknown risk type {risk.type} for risk {risk.name}")
 
+    validate_lookup_configuration(builder, risk)
+
+
+def validate_lookup_configuration(builder, risk: EntityString) -> None:
+    # Validate the configuration for the distribution type
+    distribution_type = get_distribution_type(builder, risk)
+    config = builder.configuration[risk.name]
+    weights_columns = set(
+        config["ensemble_distribution_weights"]["categorical_columns"]
+        + config["ensemble_distribution_weights"]["continuous_columns"]
+    )
+    mean_columns = set(
+        config["exposure"]["categorical_columns"] + config["exposure"]["continuous_columns"]
+    )
+    sd_columns = set(
+        config["exposure_standard_deviation"]["categorical_columns"]
+        + config["exposure_standard_deviation"]["continuous_columns"]
+    )
+    if distribution_type == "ensemble":
+        # mean, sd, and distribution weights must all be the same
+        if not weights_columns == mean_columns == sd_columns:
```

Comment:
This requirement actually needs to be even stricter - we need the categorical columns to match and the continuous columns to match, e.g. we can't have sex as a categorical column for weights and a continuous column for mean.